l5kit/l5kit/rasterization/satellite_rasterizer.py

```diff
@@ -68,7 +102,7 @@ def rasterize(
         sat_translation = transform_point(np.append(world_translation, ego_translation[2]), self.map_to_sat)
 
         # Note 1: there is a negation here, unknown why this is necessary.
-        # My best guess is because Y is flipped, maybe we can do this more elegantly.
+        # Note 2 (lberg): my best guess is that cv2 rotation are clockwise in fact.
```

Comment:
Isn't this the same negation as the one we do for agents rasterizer etc? My explanation is that the rotation is counterclockwise and this should be negative to align the world with the ego yaw, as you are doing the inverse rotation from rotating ego by ego_yaw. As we are working in 2D, inverse rotation is the same as rotation by negative angle. 

If we do negate twice, then yes, this might be to accommodate for cv2 rotation to be clockwise, but iirc that is not the case.

Reply:
nope, we don't negate anymore in `BoxRasterizer` (see [here](https://github.com/lyft/l5kit/blob/eaf31c935c7389e1e4b3cf8579a0d65ba07a6fbe/l5kit/l5kit/rasterization/box_rasterizer.py#L108)). Also, cv2 rotations are counterclock in a left-system, but we're in a right one here, so we need to compensate that. That's my best explanation right now, but I'm not claiming it's the right one xD

Reply:
Yes we do, here: https://github.com/lyft/l5kit/blob/e8018a5c3a11b7bde1e4e99f8e7cee3b0f3b0da0/l5kit/l5kit/geometry/transform.py#L67

So I still think this should be negated if we were in right hand system looking down the negative z-axis rotating counter-clockwise. So I wouldn't blame the minus sign here on cv2 as it should be here. Not sure how using the cv2 method manifests itself, but there should be a minus somewhere if using counterclockwise rotations. If the comment with clockwise is correct, then there actually shouldn't be a minus here.